client/src/app/gui/payload_type.rs

```diff
@@ -0,0 +1,227 @@
+use crate::app::gui::payload_type::MediaType::{Audio, AudioOrVideo, Video};
+
+#[derive(Debug)]
+pub struct PayloadType {
+    pub id: u8,
+    pub name: String,
+    pub clock_rate_in_hz: Option<u32>,
+    pub media_type: MediaType,
+}
+
+#[derive(Debug)]
+pub enum MediaType {
+    Audio,
+    Video,
+    AudioOrVideo,
+}
+
+impl PayloadType {
+    pub fn new(payload_type: u8) -> Self {
+        match payload_type {
+            0 => PayloadType {
+                id: 0,
+                name: "PCMU".to_string(),
+                clock_rate_in_hz: Some(8000),
+                media_type: Audio,
+            },
+            1 => PayloadType {
+                id: 1,
+                name: "reserved (previously FS-1016 CELP)".to_string(),
+                clock_rate_in_hz: Some(8000),
+                media_type: Audio,
+            },
```

Comment:
The amount of repetition here is huge. Have you considered making a vector with tuples, similar to `rtp_packets_table.rs` in order not to repeat the field names (all with the assumption that we'll stick with the struct).

Reply:
vector would be more concise, but I don't like the fact that I would need to create vector and then find in it the specific payload type i am looking for

Reply:
```rust
impl PayloadType {
    pub fn new(payload_type: u8) -> Self {
        let payload_types = [
            ("PCMU", Audio, 8000),
            ("reserved (previously FS-1016 CELP)", Audio, 8000),
            ("reserved (previously FS-1016 CELP)", Audio, 8000),
            ("GSM", Audio, 8000),
            ("G723", Audio, 8000),
            ("DVI4", Audio, 8000),
            ("DVI4", Audio, 16_000),
            ("LPC", Audio, 8000),
            ("PCMA", Audio, 8000),
            ("G722", Audio, 8000),
            ("L16", Audio, 44_100),
            ("L16", Audio, 44_100),
            ("QCELP", Audio, 8000),
            ("CN", Audio, 8000),
            ("MPA", Audio, 90_000),
            ("G728", Audio, 8000),
            ("DVI4", Audio, 11_025),
            ("DVI4", Audio, 22_050),
            ("G729", Audio, 8000),
            ("reserved (previously CN)", Audio, 0),
        ];

        let (name, media_type, freq) = match payload_type {
            0..=19 => payload_types[payload_type],
            25 => ("CELLB", Video, 90_000),
            26 => ("JPEG", Video, 90_000),
            28 => ("nv", Video, 90_000),
            31 => ("H261", Video, 90_000),
            32 => ("MPV", Video, 90_000),
            33 => ("MP2T", AudioOrVideo, 90_000),
            34 => ("H263", Video, 90_000),
            72..=76 => ("reserved", AudioOrVideo, 0),
            77..=79 => ("unassigned", AudioOrVideo, 0),
            _ => ("dynamic", AudioOrVideo, 0),
        };

        PayloadType {
            id: payload_type,
            name,
            clock_rate_in_hz: freq,
            media_type
        }
    }
}
```

Something like that is infinitely more concise, less repetitive.